api/authorization/namespace_permissions.go

```diff
@@ -94,3 +97,43 @@ func (o *NamespacePermissions) AuthorizedIn(ctx context.Context, identity Identi
 
 	return false, nil
 }
+
+func (o *NamespacePermissions) hasRole(ctx context.Context, info Info, namespacedName types.NamespacedName) (bool, error) {
+	identity, err := o.identityProvider.GetIdentity(ctx, info)
+	if err != nil {
+		return false, err
+	}
+
+	roleBinding := &rbacv1.RoleBinding{}
+	err = o.privilegedClient.Get(ctx, namespacedName, roleBinding)
+	if err != nil {
+		return false, err
+	}
+
+	for _, subj := range roleBinding.Subjects {
+		if subj.Name == identity.Name && subj.Kind == identity.Kind {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (o *NamespacePermissions) AuthorizedCreateOrg(ctx context.Context, info Info) (bool, error) {
+	hasRole, err := o.hasRole(ctx, info, types.NamespacedName{Namespace: "cf", Name: "default-admin-binding"})
+	if hasRole {
+		return true, err
+	}
+	if k8serrors.IsNotFound(err) {
+		return false, nil
+	}
+	hasRole, err = o.hasRole(ctx, info, types.NamespacedName{Namespace: "cf", Name: "korifi-controllers-organization-manager"})
```

Comment:
It feels off to have hardcoded role names and namespaces here. I know `RootNamespace` is definitely a config property and I think this `RoleMappings` thing is something that we could use as well. https://github.com/cloudfoundry/korifi/blob/main/api/config/config.go#L42

Reply:
Forgot to go back to change that :sweat_smile:. I'll take a loot at the RoleMappings.

Reply:
What if you just called `AuthorizedIn` for the user on the root namespace? Would that be sufficient to prevent the admin user from getting into this situation?

Reply:
We can use RoleMappings to keep names more consistent, but unless we check the actual RBAC perms on a role, we wouldn't know anything about the RBAC associated with a custom role. That means listing all RoleBindings in the root namespace, filtering for the User/ServiceAccount, then fetching all Roles/ClusterRoles, then iterating through all rules to find out if they have CFOrg create. This would be more flexible, but much more heavy.

Reply:
I don't think it matters if they have `CFOrg` create or not, the call on line 94 to `r.createOrgCR(ctx, info, userClient, &korifiv1alpha1.CFOrg{ ...` will fail if they don't have permission to create orgs. We're effectively doing double authorization with this.

What I'm suggesting is just calling `AuthorizedIn` as is. That would list all the RoleBindings and filter for the User/ServiceAccount sure, but that's still lighter-weight than all of our list endpoints.

Reply:
> What if you just called `AuthorizedIn` for the user on the root namespace? Would that be sufficient to prevent the admin user from getting into this situation?

We could leverage some of the existing functionality for this purpose, but it would solve the issue by side effect rather than intentionally.